File: ts/examples/whois.ts
import { Client, jsonReplacer } from "../src/index.js";

const addr = process.argv[2];
if (!addr) {
  console.error("usage: whois <addr>");
  process.exit(1);
}

const client = new Client();
const result = await client.whoIs(addr);
console.log(JSON.stringify(result, jsonReplacer, 2));
client.destroy();

File: python/examples/whois.py
"""Look up the owner of an IP address and print as JSON."""

import json
import sys

import msgspec

from tslocal import Client

if len(sys.argv) < 2:
    print(f"usage: {sys.argv[0]} <addr>", file=sys.stderr)
    sys.exit(1)

with Client() as client:
    resp = client.who_is(sys.argv[1])
    json.dump(msgspec.to_builtins(resp), sys.stdout, indent=2)
    sys.stdout.write("\n")
